metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -728,6 +775,10 @@ class MetalsLanguageServer(
       event.eventType() match {
         case EventType.CREATE =>
           buildTargets.onCreate(path)
+
+          // Modifying the file in this place using applyEdit request can cause race condition
+          // File can be modified after receiving didOpen notification
+          createdFiles.add(path)
```

Comment:
We should check if the file is a Scala file - I think it's working now, since we are not applying it on JAva files, but I think it's being added unnecessarily to the set.

Reply:
+1 to the `isScala` variant. This would nicely fit with the already present `isScalaOrJava`